cmd/cli/serve/serve.go

```
@@ -177,23 +177,37 @@ func GetPeers(OS *ServeOptions) ([]multiaddr.Multiaddr, error) {
 	return peers, nil
 }
 
-func GetComputeConfig(OS *ServeOptions) node.ComputeConfig {
+func GetComputeConfig(OS *ServeOptions) (node.ComputeConfig, error) {
+	totalResourcLimitsConfig := models.ResourcesConfig{
+		CPU:    OS.LimitTotalCPU,
+		Memory: OS.LimitTotalMemory,
+		GPU:    OS.LimitTotalGPU,
+	}
+
+	jobResourcesLimitsConfig := models.ResourcesConfig{
+		CPU:    OS.LimitJobCPU,
+		Memory: OS.LimitJobMemory,
+		GPU:    OS.LimitJobGPU,
+	}
+
+	parsedTotalLimits, err := totalResourcLimitsConfig.ToResources()
+	if err != nil {
+		return node.ComputeConfig{}, err
+	}
+
```

we used to silently ignore resource parsing errors